src/mongoose_hooks.erl

```diff
@@ -582,15 +541,93 @@ roster_groups(Server, Acc) ->
 %%% * RemoteBareJID, a bare JID of the other user.
 %%%
 %%% The arguments and the return value types correspond to the following spec.
--spec roster_get_jid_info(HookServer, InitialValue, LUser, RemoteJID) -> Result when
-      HookServer :: jid:lserver(),
+-spec roster_get_jid_info(LServer, InitialValue, LUser, RemoteJID) -> Result when
+    LServer :: jid:lserver(),
       InitialValue :: {mod_roster:subscription_state(), []},
       LUser :: jid:luser(),
       RemoteJID :: jid:jid() | jid:simple_jid(),
       Result :: {mod_roster:subscription_state(), [binary()]}.
-roster_get_jid_info(HookServer, InitialValue, LUser, RemBareJID) ->
-    ejabberd_hooks:run_fold(roster_get_jid_info, HookServer, InitialValue,
-                            [LUser, HookServer, RemBareJID]).
+roster_get_jid_info(LServer, InitialValue, LUser, RemBareJID) ->
+    ejabberd_hooks:run_fold(roster_get_jid_info, LServer, InitialValue,
+                            [LUser, LServer, RemBareJID]).
+
+%%% @doc The `roster_get_subscription_lists' hook is called to extract user's subscription list.
+-spec roster_get_subscription_lists(Server, Acc, User) -> Result when
+    Server :: jid:server(),
+    Acc ::mongoose_acc:t(),
+    User :: jid:user(),
+    Result :: mongoose_acc:t().
+roster_get_subscription_lists(Server, Acc, User) ->
+    ejabberd_hooks:run_fold(roster_get_subscription_lists, Server, Acc, [User, Server]).
+
+%%% @doc The `roster_get_versioning_feature' hook is called to determine if roster versioning is enabled.
+-spec roster_get_versioning_feature(Server, Acc) -> Result when
+    Server :: jid:server(),
+    Acc :: [exml:element()],
+    Result :: [exml:element()].
+roster_get_versioning_feature(Server, Acc) ->
+    ejabberd_hooks:run_fold(roster_get_versioning_feature,
+                            Server, Acc, [Server]).
+
+%%% @doc The `roster_in_subscription' hook is a subscription presence is routed to a user.
+-spec roster_in_subscription(LServer, Acc, User, Server, From, Type, Reason) -> Result when
+    LServer :: jid:lserver(),
+    Acc :: mongoose_acc:t(),
+    User :: jid:user(),
+    Server :: jid:server(),
+    From :: jid:jid(),
+    Type :: mod_roster:sub_presence(),
+    Reason :: any(),
+    Result :: mongoose_acc:t().
+roster_in_subscription(LServer, Acc, User, Server, From, Type, Reason) ->
+    ejabberd_hooks:run_fold(
+        roster_in_subscription,
+        LServer,
+        Acc,
+        [User, Server, From, Type, Reason]).
+
+%%% @doc The `roster_out_subscription' hook is a user sends out subsciption.
```

Comment:
Minor: missing 'called to determine if'

Reply:
Oops, missed it. Is the sentence itself correct?